facebookresearch/ParlAI
# Add Amazon QA Dataset

**Patch description**
Adds the Amazon QA dataset to the tasks.
Link: [http://jmcauley.ucsd.edu/data/amazon/qa/](url)

**Testing steps**
```
python examples/display_data.py -t amazon_qa

- - - - - - - - - - - - - - - - - - - - -
~~
[asin]: B00004U9JO
[questionType]: open-ended
[questionTime]: N/A
[askerID]: N/A
[answerType]: N/A
[answerTime]: Mar 9, 2014
[unixTime]: 1394352000
[answererID]: N/A
[helpful]: N/A
[answerScore]: N/A
[amazon_qa]: I have a badger 5 it is leaking water at bottom near reset bottom plus does a new one came with a ss ring for inside the sink
[labels: If your garbage disposal is leaking out the bottom near the reset button, you most likely have a leak in the metal plate that is located directly under the blades. This plate is designed to protect the electrically powered parts of the motor (the rotor, the shaft, etc.) from moisture, but it is rather thin and is only made of galvanized steel. After a year or two, especially if you use highly acidic cleaners which you rinse down your sink, the zinc layer will corrode and expose the steel, which will then also start to corrode, until something hard eventually pokes some kind of hole through the plate somewhere (could be somewhere very small), signaling the end of the machine's service life. The plate is not meant to be repaired, only replaced along with the rest of the unit. That water is getting through to your rotor is an electrical hazard, and should be dealt with immediately (by removing the damaged unit). The manufacturer has figured out that the blades need to me made of stainless steel, but has chosen to skimp on making other interior parts of this more-expensive metal. Good luck!]
- - - - - - - - - - - - - - - - - - - - -
~~
[asin]: B00004U9JO
[questionType]: open-ended
[questionTime]: N/A
[askerID]: N/A
[answerType]: N/A
[answerTime]: Sep 24, 2014
[unixTime]: 1411542000
[answererID]: N/A
[helpful]: N/A
[answerScore]: N/A
[amazon_qa]: If no power cord is included, how do I plug in the disposal unit?
[labels: I ttok mine off of the broken unit and installed it on the new one.]
- - - - - - - - - - - - - - - - - - - - -
~~
[asin]: B00004U9JO
[questionType]: open-ended
[questionTime]: N/A
[askerID]: N/A
[answerType]: N/A
[answerTime]: Nov 10, 2013
[unixTime]: 1384070400
[answererID]: N/A
[helpful]: N/A
[answerScore]: N/A
[amazon_qa]: Can someone tell me if this Badger 5 disposer is simple to replace (with my old broken Badger 5)? (i.e. Can I do it myself vs. hiring a plumber )
[labels: I replaced my a couple of months ago, I'm semi-handy fixing a lot of small stuff around the house. I'm small and in my late seventies, the only problem was holding the unit up in place while turning the locking ring. I no longer have the upper body strength to do it so I used a stack of books. Made sure all the connections were tight and have had so problems since. It clean and in good shape so I left the old badger mount in the sink. Took about forty min, but a lot of that was getting the books.]
- - - - - - - - - - - - - - - - - - - - -
~~
[asin]: B00004U9JO
[questionType]: open-ended
[questionTime]: N/A
[askerID]: N/A
[answerType]: N/A
[answerTime]: Jul 24, 2013
[unixTime]: 1374649200
[answererID]: N/A
[helpful]: N/A
[answerScore]: N/A
[amazon_qa]: Is this Badger 5 compatible with the Badger 5 model 5-81? Same size and fittings?
[labels: It fit the same as the one we replaced]
- - - - - - - - - - - - - - - - - - - - -
~~
[ loaded 5334606 episodes with a total of 5334606 examples ]
```

**Data tests (if applicable)**
If you added a new teacher, you will be asked to run
`python setup.py test -s tests.suites.datatests`. Please paste this log here.
```
[ loaded 5334606 episodes with a total of 5334606 examples ]
2079s elapsed: {'exs': 5334606, '%done': '100.00%', 'time_left': '0s', 'missing_text': 0, 'missing_labels': 0, 'missing_label_candidates': 5334606, 'empty_string_label_candidates': 0, 'label_candidates_with_missing_label': 0, 'did_not_return_message': 0}
```
